### PR TITLE
Adding `neuralforecast` from PyPI

### PR DESCRIPTION
**Pull request by sugatoray**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by sugatoray** [comment]
## Pip Check failed: Reason

[(error)](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=478796&view=logs&j=6f142865-96c3-535c-b7ea-873d86b887bd&t=22b0682d-ab9e-55d7-9c79-49f3c3ba4823&l=1914)

```sh
import: 'neuralforecast'
+ pip check
pytorch-lightning 1.5.10 has requirement pyDeprecate==0.3.1, but you have pydeprecate 0.3.2.
py7zr 0.18.1 has requirement pyppmd<0.18.0,>=0.17.0, but you have pyppmd 0.18.0.
py7zr 0.18.1 has requirement zipfile-deflate64>=0.2.0, but you have zipfile-deflate64 0.0.0.
Tests failed for neuralforecast-0.0.7-pyhd8ed1ab_0.tar.bz2 - moving package to /home/conda/staged-recipes/build_artifacts/broken
WARNING:conda_build.build:Tests failed for neuralforecast-0.0.7-pyhd8ed1ab_0.tar.bz2 - moving package to /home/conda/staged-recipes/build_artifacts/broken
TESTS FAILED: neuralforecast-0.0.7-pyhd8ed1ab_0.tar.bz2
##[error]Bash exited with code '1'.
```

**Comment by sugatoray** [comment]
## Build Fails at pip-check

[(error)](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=479656&view=logs&j=6f142865-96c3-535c-b7ea-873d86b887bd&t=22b0682d-ab9e-55d7-9c79-49f3c3ba4823&l=1884)

```sh
import: 'neuralforecast'
+ pip check
py7zr 0.18.1 has requirement zipfile-deflate64>=0.2.0, but you have zipfile-deflate64 0.0.0.
Tests failed for neuralforecast-0.0.7-pyhd8ed1ab_0.tar.bz2 - moving package to /home/conda/staged-recipes/build_artifacts/broken
WARNING:conda_build.build:Tests failed for neuralforecast-0.0.7-pyhd8ed1ab_0.tar.bz2 - moving package to /home/conda/staged-recipes/build_artifacts/broken
TESTS FAILED: neuralforecast-0.0.7-pyhd8ed1ab_0.tar.bz2
##[error]Bash exited with code '1'.
```

**Comment by sugatoray** [comment]
## Reason for Windows build failure

PyTorch is not avilable on conda-forge for windows platform.

**Comment by sugatoray** [comment]
PR is ready. Request for review/merging.

cc: @conda-forge/help-python

File: recipes/neuralforecast/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    - python >=3.7
+    - fastcore
+    - gdown
+    - hyperopt
+    - matplotlib-base
+    - numba
+    - openpyxl
+    - pandas
+    - pip
+    - py7zr >=0.18.1
```
Comment (BastianZim) [inline recipes/neuralforecast/meta.yaml:31]: I cannot find that version specifier upstream. is that from your pip checks?

File: recipes/neuralforecast/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    - pytorch >=1.4
+    - torchinfo
+    - tqdm
+    - xlrd
+    # The following is necessary to pass pip-check
+    # pyDeprecate==0.3.1 is necessary for PL==1.5.10
+    - pyDeprecate 0.3.1
+    # pyppmd and zipfile-deflate constraints are necessary for py7zr
+    - pyppmd >=0.17.0,<0.18.0
+    - zipfile-deflate64 >=0.2.0
```
Comment (BastianZim) [inline recipes/neuralforecast/meta.yaml:45]: Can you please add an issue in the `py7zr` feedstock to have this updated or make a PR there yourself?

**Review by BastianZim — CHANGES_REQUESTED** [review]
Just a couple of things then this is ready.

File: recipes/neuralforecast/meta.yaml (hunk 1 above)
Comment (sugatoray) [reply recipes/neuralforecast/meta.yaml:31]: Yes, pip check was showing that. I am quite sure, if pip check is dropped, that version `>=0.18.1` will not be necessary.

**Comment by sugatoray** [comment]
@BastianZim Thank you.

**Pull request merged**